scripts/js/lib/markdownTitles.ts

```diff
@@ -0,0 +1,47 @@
+import { unified } from "unified";
+import remarkParse from "remark-parse";
+import remarkGfm from "remark-gfm";
+import remarkFrontmatter from "remark-frontmatter";
+import { visit } from "unist-util-visit";
+import { Root } from "mdast";
+import yaml from "js-yaml";
+
+export async function collectHeadingTitleMismatch(markdown: string): Promise<string[]> {
+  const mismatches: string[] = [];

```

Comment:
Can you use a `Set` rather than an array to match the images check? Then you'd just do `.add()` rather than `.push()` on line 41